### ISSUE TITLE
URGENT: Twitter Cards Broken for Demo

### ISSUE THREAD

**Issue by anexact** [usergroupcoop/msf-reach#93]
When we make a new event and issue a Report Card link, it works if it is sent over WhatsApp or over email; however, Report Cards break when link is send via Twitter.

I noticed that there is a difference in the url; in the link via WA/email, it is this:
<img width="975" alt="screen shot 2017-08-24 at 12 20 51 pm" src="https://user-images.githubusercontent.com/20785542/29649395-60c3a33c-88c6-11e7-92f1-6343387852e5.png">

Via Twitter, there is no - at the end of the url:
<img width="904" alt="screen shot 2017-08-24 at 12 21 06 pm" src="https://user-images.githubusercontent.com/20785542/29649413-71ad599a-88c6-11e7-9061-f7ae1651539a.png">

We also tried manually adding the dash to the url, but this also gets the following error message:

<img width="952" alt="screen shot 2017-08-24 at 12 28 02 pm" src="https://user-images.githubusercontent.com/20785542/29649448-b130b210-88c6-11e7-8cca-c4adbd054575.png">

This is the most basic functional element for workshop demo; please have a look at it because if this element doesn't work, the demo will be cancelled.

**Comment by mehrdadgit** [comment]
Hi @anexact  

I had a look, this is kinda twitter bug and it happens only in the not-that-often case the auto generated reportkey has a - at the end; you check all other events, with their reportkey not having a dash ->  works in twitter too. 

I don't want to change the API at this late stage.  But this will be an easy fix.

For now, I don't see a reason adding the dash manually to the URL wouldn't work ... (looking at your last screenshot, the URL does not seem to have a dash, therefore failing. ..

**Comment by anexact** [comment]
Unfortunately, the problem cannot be solved by manually adding a "-" to the report.
If you click the report card link from Twitter, it opens like this:
  
<img width="749" alt="screen shot 2017-08-24 at 2 35 06 pm" src="https://user-images.githubusercontent.com/20785542/29652630-eb0711e8-88d8-11e7-9187-f2c276df3d0f.png">

Then, if you add a "-" to the url and fill it in, you still get this error:
<img width="779" alt="screen shot 2017-08-24 at 2 36 37 pm" src="https://user-images.githubusercontent.com/20785542/29652645-fcac2744-88d8-11e7-8a95-0ceb9549c90d.png">

Notably, if you open the Report card link from an email or WA, the url is different in the Chrome browser:

<img width="719" alt="screen shot 2017-08-24 at 2 36 56 pm" src="https://user-images.githubusercontent.com/20785542/29652668-1de2d502-88d9-11e7-82d1-c213015c206d.png">

You can see that the "Secure" Lock and the https are in Green, whereas in Twitter they are not. 

Thoughts?

**Comment by matthewberryman** [comment]
I’ll fix this when I get into the hotel tonight. Needs one line of the code changed but also I need to update the database too.

Sent from my iPhone

On 24 Aug 2017, at 14:35, Etienne Turpin <notifications@github.com<mailto:notifications@github.com>> wrote:


Unfortunately, the problem cannot be solved by manually adding a "-" to the report.
If you click the report card link from Twitter, it opens like this:

[screen shot 2017-08-24 at 2 35 06 pm]<https://user-images.githubusercontent.com/20785542/29652630-eb0711e8-88d8-11e7-9187-f2c276df3d0f.png>

Then, if you add a "-" to the url and fill it in, you still get this error:
[screen shot 2017-08-24 at 2 36 37 pm]<https://user-images.githubusercontent.com/20785542/29652645-fcac2744-88d8-11e7-8a95-0ceb9549c90d.png>

Notably, if you open the Report card link from an email or WA, the url is different in the Chrome browser:

[screen shot 2017-08-24 at 2 36 56 pm]<https://user-images.githubusercontent.com/20785542/29652668-1de2d502-88d9-11e7-82d1-c213015c206d.png>

You can see that the "Secure" Lock and the https are in Green, whereas in Twitter they are not.

Thoughts?

**Comment by matthewberryman** [comment]
Tested fix in 41bc78f including updating existing entries in database with 
```sql
UPDATE cognicity.events SET report_key = replace(report_key, '-', '.');
UPDATE cognicity.reports SET report_key = replace(report_key, '-', '.');
```
Will restore database to state prior to testing and reapply fix, as I created a lot of test events and reports.

**Comment by matthewberryman** [comment]
Should add I tested on twitter too with the new changes in 41bc78f and that worked.
